src/suggest/date-suggest.ts

```diff
@@ -58,69 +57,99 @@ export default class DateSuggest extends EditorSuggest<IDateCompletion> {
     // });
   }
 
-  getSuggestions(context: EditorSuggestContext): IDateCompletion[] {
+  getSuggestions(context: EditorSuggestContext): string[] {
     // handle no matches
     const suggestions = this.getDateSuggestions(context.query);
-    if (suggestions.length) {
-      return suggestions;
-    } else {
-      return [{ label: context.query }];
-    }
+    return suggestions.length ? suggestions : [ context.query ];
   }
 
-  getDateSuggestions(inputStr: string): IDateCompletion[] {
-    if (inputStr.match(/^time/)) {
-      return ["now", "+15 minutes", "+1 hour", "-15 minutes", "-1 hour"]
-        .map((val) => ({ label: `time:${val}` }))
-        .filter((item) => item.label.toLowerCase().startsWith(inputStr));
+  getDateSuggestions(inputStr: string): string[] {
+    return this.unique(this.plugin.settings.languages.flatMap(
+      language => {
+        let suggestions = this.getTimeSuggestions(inputStr, language);
+        if (suggestions)
+          return suggestions;
+        suggestions = this.getImmediateSuggestions(inputStr, language);
+        if (suggestions)
+          return suggestions;
+
+        suggestions = this.getRelativeSuggestions(inputStr, language);
+        if (suggestions)
+          return suggestions;
+
+        return this.defaultSuggestions(inputStr, language);
+      }
+    ));
+  }
+
+  private getTimeSuggestions(inputStr: string, lang: string): string[] {
+    if (inputStr.match(new RegExp(`^${t("time", lang)}`))) {
```

Comment:
This is the way that I dealt with the suggestions: the regex contains translatable parts. So `time` for English, `tijd` voor Dutch, etc. Then everything about translations (both the _actual_ suggestions, and the regex for determining them) was done the same way